eq_cir_proxy_service/services/instrument/instrument_conversion_service.py

```diff
@@ -0,0 +1,94 @@
+"""This module requests conversion of the instrument from Converter Service."""
+
+import os
+
+from fastapi import HTTPException, status
+from httpx import AsyncClient, RequestError
+from semver import Version
+
+from eq_cir_proxy_service.config.logging_config import logging
+from eq_cir_proxy_service.exceptions import exception_messages
+from eq_cir_proxy_service.types.custom_types import Instrument
+
+logger = logging.getLogger(__name__)
+
+
+async def convert_instrument(instrument: Instrument, target_version: str) -> Instrument:
+    """Requests conversion of the instrument from Converter Service.
+
+    Parameters:
+    - instrument: The instrument.
+    - current_version: The current version of the instrument.
+    - target_version: The target version of the instrument.
+
+    Returns:
+    - dict: The converted instrument.
+    """
+    if not instrument.get("validator_version"):
+        logger.error("Instrument version is missing")
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail={"status": "error", "message": exception_messages.EXCEPTION_400_INVALID_INSTRUMENT},
+        )
+
+    current_version: str = str(instrument["validator_version"])
+    parsed_current_version = Version.parse(current_version)
```

Comment:
I'm not saying we want to but just wondering, do we want to try/except the version parsing (both in one `try`?) in case it is a bad version? Similar to how you raise exception on failed `get`?

Reply:
I like the way you approached it (reusability etc).